File: MATLAB/writeFixedPointBinary.m
function writeFixedPointBinary(signal, filename, word_length, fraction_length)
    % Write fixed-point signal as binary s16.15 to text file
    % Each line contains one 16-bit binary value
    
    fid = fopen(filename, 'w');
    if fid == -1
        error('Cannot open file %s for writing', filename);
    end
    
    % Convert to fixed-point if not already
    if ~isfi(signal)
        signal_fi = fi(signal, 1, word_length, fraction_length);
    else
        signal_fi = signal;
    end
    
    % Write each sample as binary string
    for i = 1:length(signal_fi)
        % Get binary representation
        bin_str = bin(signal_fi(i));
        
        % Ensure it's exactly 16 bits (pad with zeros if needed)
        if length(bin_str) < word_length
            bin_str = [repmat('0', 1, word_length - length(bin_str)), bin_str];
        elseif length(bin_str) > word_length
            bin_str = bin_str(end-word_length+1:end);
        end
        
        fprintf(fid, '%s\n', bin_str);
    end
    
    fclose(fid);
end

File: Python_Script/write_fixed_point_binary.py
import os
from fixed_point_utils import *

def write_fixed_point_binary(signal_data, filename, word_length = 16, fraction_length = 15):
    """
    Write fixed-point signal as binary to text file.
    Each line contains one 16-bit binary value.
    """
    os.makedirs(os.path.dirname(filename), exist_ok = True)

    with open(filename, 'w') as f:
        for sample in signal_data:
            fp_bin = quantize_fixed_point(sample, word_length, fraction_length, return_FP = False)
            f.write(f"{int_to_binary(fp_bin, word_length = word_length)}\n")